packages/ember-data/lib/system/store.js

```diff
@@ -860,6 +860,33 @@ Store = Ember.Object.extend({
   },
 
   /**
+    This method delegates a query to the adapter. This is the one place where
+    adapter-level semantics are exposed to the application.
+
+    Exposing queries this way seems preferable to creating an abstract query
+    language for all server-side queries, and then require all adapters to
+    implement them.
+
+    This method returns a promise, which is resolved with a `RecordArray`
+    once the server returns.
+
+    @method findQuery
+    @private
+    @param {String or subclass of DS.Model} type
+    @param {any} query an opaque query to be used by the adapter
+    @return {Promise} promise
+  */
+  findQueryOne: function(typeName, id, query) {
```

Comment:
I'd be in favor of renaming the store api to `fetchQueryOne` since it looks like this will always fetch a new record. Keeping the adapter API as `findQueryOne` seems fine and consistent with the other adapter methods.